src/containers/Dev/Snapshots.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useState, useEffect, useRef } from 'react'
+import {
+  Button,
+  Grid,
+  GridColumn,
+  Icon,
+  Input,
+  Label,
+  Loader,
+  Popup,
+  Portal,
+  Segment,
+} from 'semantic-ui-react'
+import config from '../../config'
+
+const snapshotsBaseUrl = `${config.serverREST}/snapshot`
+const snapshotListUrl = `${snapshotsBaseUrl}/list`
+const takeSnapshotUrl = `${snapshotsBaseUrl}/take`
+const useSnapshotUrl = `${snapshotsBaseUrl}/use`
+const snapshotFilesUrl = `${snapshotsBaseUrl}/files`
+const uploadSnapshotUrl = `${snapshotsBaseUrl}/upload`
+// const diffSnapshotUrl = `${snapshotsBaseUrl}/diff`
+
+const Snapshots: React.FC = () => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [compareFrom, setCompareFrom] = useState('')
+  const [isPortalOpen, setIsPortalOpen] = useState(false)
+  const [isSnapshotError, setIsSnapshotError] = useState(false)
+
+  const [data, setData] = useState<string[] | null>(null)
+
+  useEffect(() => {
+    if (isOpen) {
+      setData(null)
+      setCompareFrom('')
+      setIsSnapshotError(false)
+      getList()
+    }
+  }, [isOpen])
+
+  const getList = async () => {
+    try {
+      const snapshotListRaw = await fetch(snapshotListUrl, { method: 'GET' })
+      const snapshotList: string[] = (await snapshotListRaw.json()).snapshotsNames
+
+      setData(snapshotList)
+    } catch (e) {}
+  }
+
+  const normaliseSnapshotName = (name: string) =>
+    // not word, not digit
+    name.replace(/[^\w\d]/g, '')
+
+  const takeSnapshot = async (name: string) => {
+    if (!name) return
```

Comment:
The back-end handles no parameter and sets a default "current" -- so we should allow it to be run with no input name.

Reply:
I thought it was a bit weird to takeSnapshot with no name, and have it create/overwrite the one named 'current' ?

Reply:
Sure, then maybe the front-end UI could insert a default name in the input field?

Reply:
could do, current placeholder is 'New snapshot' to indicate what input is related to, and if we do default value it would appear instead of placeholder, so not sure..